File: src/Orchestrator/src/Application/Admin/Queries/GetUserByEmail/GetUserByEmailRvm.cs
namespace Application.Admin.Queries.GetUserByEmail;

public class GetUserByEmailRvm
{
    public required string UserId { get; init; }
    public required string SignerAddress { get; init; }
    public required string WalletAddress { get; init; }
    public required bool EmailConfirmed { get; init; }
}

File: src/frontend/lib/admin/models/vm/get_user_by_email_rvm.dart
class GetUserByEmailRvm {
  final String userId;
  final String walletAddress;
  final String signerAddress;
  final bool emailConfirmed;

  GetUserByEmailRvm.fromMap(Map<String, dynamic> map)
      : userId = map['userId'],
        walletAddress = map['walletAddress'],
        signerAddress = map['signerAddress'],
        emailConfirmed = map['emailConfirmed'];
}
